staging/src/k8s.io/apiserver/pkg/util/flowcontrol/borrowing_test.go

```diff
@@ -223,12 +221,6 @@ func TestBorrowing(t *testing.T) {
 
 			// Do the checking
 
-			t.Log("waiting for the controller Run function to shutdown gracefully")
-			controllerErr := <-controllerCompletionCh
-			close(controllerCompletionCh)
-			if controllerErr != nil {
-				t.Errorf("expected nil error from controller Run function, but got: %#v", controllerErr)
-			}
 			if results0.Average < 15.5 || results0.Average > 16.1 {
```

Comment:
@tkashem @MikeSpreitzer this is the only place where I could see how asynchronous shutdown leads to different results. If I am not mistaken though we are talking here about 15s vs. milliseconds. Plus, **the synchronous shutdown was not working before either**, and it didn't flake. Hence, my comment that making this synchronous is out of scope of this PR.

Reply:
@sttts: I am not sure I understand your comment. Looking at the earlier lines of code in this test, we see a sleep of 15s followed by the assignment to `results0`. All that is before starting to shut down the server.  So whether the server shutdown is synchronous or not seems irrelevant?

Reply:
Somewhere @tkashem mentioned that synchronous shutdown would help to avoid flakes. I tried to find where that could be the case. And as you say, here it is about milliseconds whether it is sync or async, and within 15s it should not make a difference.

Reply:
I am making a stronger statement, with a little narrower scope. This particular test evaluates stats that are collected before shutdown even starts, so the success of this test is completely independent of how shutdown goes.

OTOH, success of _the next_ test will depend on whether this test's shutdown is complete, and that is the generic issue that @tkashem is talking about.

Reply:
Resolved.